---
title: "Assignment 1"
author: "Bindoff, A."
output: html_document
bibliography: ass_1.bib
---

`r Sys.time()`

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
library(dplyr)
library(ggplot2)
library(knitr)
library(readr)
library(seqinr)
library(bio3d)
set.seed(42)

```

# Background


```{r, echo = FALSE}
sq <- as.vector(read.table("seq.txt")$V1)
```

The BLASTn algorithm [@altschul1997gapped] was used to find high-scoring segment pairs (HSP) for the nucleotide query sequence `r sq` against the database maintained at [NCBI](https://blast.ncbi.nlm.nih.gov/). An API call was made initially using the R package `annotate` [@gentleman2018], however a query using the website interface proved to be much faster. Results were saved by copying and pasting the resulting table into Excel and saving as a .csv file.  

```{r, eval = FALSE}
#biocLite("annotate")
library(annotate)
bs <- blastSequences(sq, timeout = 480, as = "data.frame")
```

## BLAST Results

```{r}
bs <- read_csv("bs.csv")

dplyr::select(bs, Description, `E value`, `Per. Ident`, `Query Cover`) %>%
  dplyr::arrange(`E value`) %>%
  top_n(24) %>%
  kable()

```

```{r}
ggplot(bs, aes(x = `E value`, fill = factor(substr(Description, 1, 5)))) + 
  geom_histogram()

```

### Query cover

Query cover is the proportion of characters of the query sequence that are included in the [aligned] target sequence (x100).


### Percent identity

Percent identity describes the proportion of characters in the query sequence that match the target sequence (x100), such that identical sequences will have a percent identity of 100% and sequences with no matching characters will have a percent identity of 0%.  

This number is not adjusted for the length of the query or target sequences, nor does it take into account the probability of matching by chance.  


### *E*-value

#### Pairwise *E*-value

Consider the comparison of two sequences. Interest lies in determining whether there is evidence of shared ancestry (homology) between the sequences. A local alignment comprises an equal length segment from each sequence. Sufficient similarity between the aligned segments may be evidence of homology. However similarity can occur by chance so it is helpful to estimate the likelihood of observing the alignment by chance alone, using a model of random sequences [@altschul1990basic].  

For sufficiently large random sequences of lengths $m$ and $n$, the expected number (*E*-value) of local alignments with score at least $S$ is given by  

\[E \approx K mn e^{-\lambda S}\]

where $K$ and $\lambda$ are natural scales for the search space size and the scoring system respectively [@altschul1990basic]. Thus, *E*-value gives a better measure of the **significance** of a match between query and target sequence than percent identity, as it makes a correction for the search space.  

```{r}
S <- seq(0, 10, by = 0.5)
E <- 2^-S
plot(E ~ S, xlab = "S'", type = "l", yaxt = "n", xaxt= "n")
```

The probability of finding exactly $z$ HSPs with score $\geq S$ is Poisson distributed  
\[e^{-E} {E^z \over z!}\]  

because the probability of finding $z = 0$ HSPs with score $\geq S$ is $e^{-E}$, the probability of finding at least one HSP with score $\geq S$ is  
\[P = 1- e^{-E}\]

For small $E$, *P*-values and *E*-values are approximately equivalent. *E*-values are reported in BLAST results because they are intuitively easier to grasp.  

#### Database *E*-values

NCBI BLAST treats the database of sequences as a single very long sequence of length $N$, and calculate *E*-values using this in the place of $n$ in the above equation. Thus, the *E*-values reported in the table above describe how many times you would expect a match by chance in a database containing all of the nucleotide sequences stored in the NCBI database.  

## Alignment

16 sequences were chosen from different species and downloaded. They were then aligned using muscle [@muscle] in Seaview [@gouy2009seaview].

```{r}
aligned <- bio3d::read.fasta("fasta_15.fst")
print.fasta(aligned)

```

Comment on difference between this alignment and BLAST alignment to query sequence  

## Neighbour Joining

SplitsTree4 [@huson2005application] was used to build neighbour-joining trees, using 1000 bootstrap replicates to estimate robustness of the tree.  

![Neighbour joining tree with bootstrap estimates](boot_strapped_nj_tree.png)

## Maximum Parsimony Tree

PHYLIP v3.696 [@felsenstein2004phylip] was used to compute parsimony trees in SeaView [@gouy2009seaview].  

![Parsimony tree with bootstrap estimates](fasta_15-Dnapars.png)

## Maximum Likelihood Tree

```{r, eval = FALSE}
# I didn't have room for Linux VM on home computer where I was doing this
cmd <- "iqtree-1.6.10-Windows\\bin\\iqtree -s fasta_15.nxs -bb 1000 -nt 2"
system(cmd)

```

```{r}
cat(read_file("fasta_15.nxs.log"))
```
```{r}
cat(read_file("fasta_15.nxs.contree"))

```

## Phylogenetic Network

HKY distances were used to estimate a phylogenetic network in SplitsTree4 [@huson2005application].  

![Phylogenetic network using HKY distances, with bootstrap estimates](hky_phylo_net.png)

# References


