<!DOCTYPE html>
<!-- saved from url=(0094)file:///C:/Users/bindoffa/AppData/Local/Temp/Temp1_Assignment%201.html.zip/Assignment%201.html -->
<html><head><meta http-equiv="Content-Type" content="text/html; charset=windows-1252">
</head>
<body style="font-family: arial, helvetica, sans-serif;font-size: 12px;">
<p>This assignment is in several parts. Each is relatively straightforward, but do feel free to ask if you want some help.</p>
<p><strong>If you have any questions&nbsp;please let me know.</strong></p>
<p>The assignment is to do with sequence comparison, alignment, and phylogenetic estimation.</p>
<p>Begin with the following sequence:</p>
<pre>AGGTGGCTCTAGAGAATCCCCAAATGCAAAACTCAGAGATCAGCAAGCAGCTGGGGTACCAGTGGAAAATGCTTACAGAAGCCGAAAAATGGCCATTCTTCGAGGAGGCACAGAGACTACAGGCCGTGCACCGAGAGAAATACCCAGACTATAAATATCGACCTCCCCCCCCGTCGGAAGGCCCTGCCACAGAAAAGTGATAAATTGCTACCTGCAGCCTCCTCCTCCCTGCTATGCAGGCAGGTGCTCGTGGATAAGTGGTATCCCCTTCACGTACAGGGACAGAAGCTCCCTCACAACCCGTGAACATAGCCAACTCGCTGCTTTGACGCAACAGGAGCACCACTACAGCTCTACAAGCCTCCGTGGCAGTCCGGAAACCTTGGCTACGCATCTGTCCGCTGACCCTCCCTTTTACCCTAAGTAACAGCTGGGACGACTTTCTGATGCTTATTTTCCATGGTGATTTCCTTACTCTTAA</pre>
<ol>
<li>Using NCBI's <strong><span style="font-family: &#39;courier new&#39;, courier, sans-serif;">BLAST</span></strong>, find out what species and gene this sequence is from (if you can... I kind of edited it a bit). Report the results of your search, with all the relevant statistics that are returned, and explain what each means.&nbsp; (E.value, percent identity, query cover etc.)</li>
<li>Make a selection of related sequences <strong>of different species</strong>: five&nbsp;hits that have the best possible E-value; five other hits in the 90-99% identity range; and five other hits with lower identity. This should give you 15 sequences. Download the complete gene if possible.</li>
<li>Align the sequences using <strong><span style="font-family: &#39;courier new&#39;, courier, sans-serif;">SeaView</span></strong><span style="font-family: &#39;courier new&#39;, courier, sans-serif;"><span style="font-family: arial, helvetica, sans-serif;"> and include your alignment in the results; then use&nbsp;</span></span>&nbsp;<strong><span style="font-family: &#39;courier new&#39;, courier, sans-serif;">muscle</span></strong>&nbsp;and save the output as a .NEXUS file, for input to phylogenetic analysis programs.&nbsp;Comment on any differences between the two alignments.</li>
<li>Build a Neighbour-Joining tree using any program you choose (cite which one; e.g., SplitsTree) for your&nbsp;alignment and estimate bootstrap values.</li>
<li>Estimate the most parsimonious tree for the sequences using any program you choose (cite which one), and save it. Also do a bootstrap analysis with at least 100 replicates to estimate the robustness of the tree you find.</li>
<li>Using IQTree (which is available in BioLinux), find the likelihood model that best fits your alignment and find the most likely tree using that model. Detail all the relevant outputs.</li>
<li>Finally, use SplitsTree and HKY distances to display a phylogenetic network for your data.</li>
<li>Describe and report all&nbsp;phylogenetic&nbsp;estimations; remark on areas of uncertainty and of robustness, and anything else that appears interesting about your findings.</li>
<li><strong>EXTENSION, if you wish:</strong> Find&nbsp;<em>another</em> gene for these species and compare your trees...</li>
</ol>
<p>Good luck, have fun!</p>
<p><strong>If you have any problems please let me know.</strong></p>

</body></html>